src/main/java/com/example/tournamentMaker/auth/AuthenticationService.java

```diff
@@ -1,2 +1,64 @@
-package com.example.tournamentMaker.auth;public class AuthenticationService {
+package com.example.tournamentMaker.auth;
+
+import com.example.tournamentMaker.address.Address;
+import com.example.tournamentMaker.config.JwtService;
+import com.example.tournamentMaker.token.Token;
+import com.example.tournamentMaker.token.TokenRepository;
+import com.example.tournamentMaker.token.TokenType;
+import com.example.tournamentMaker.user.Role;
+import com.example.tournamentMaker.user.User;
+import com.example.tournamentMaker.user.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationService {
+    private final UserRepository repository;
+    private final TokenRepository tokenRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtService jwtService;
+    private final AuthenticationManager authenticationManager;
+
+    public AuthenticationResponse register(RegisterRequest request) {
+        User user = new User(request.getFirstName(), request.getLastName(), new Address(request.getCity(),
+                request.getStreet()), request.getEmail(), request.getPassword(), Role.USER);
+        User savedUser = repository.save(user);
+        String token = jwtService.generateToken(user);
+        saveUserToken(savedUser,token);
+        return new AuthenticationResponse(token);
+    }
+
+    private void saveUserToken(User user, String token) {
+        Token tokenToSave = new Token(token, TokenType.BEARER, false, false, user);
+        tokenRepository.save(tokenToSave);
+    }
+
+    public AuthenticationResponse authenticate(AuthenticationRequest authenticationRequest) {
+        authenticationManager.authenticate(new UsernamePasswordAuthenticationToken(
+                authenticationRequest.getEmail(),authenticationRequest.getPassword()));
+
+        User user = repository.findByEmail(authenticationRequest.getEmail()).orElseThrow(); // jak nie znajdziesz to się wypierdol
```

Comment:
ten komentarz to usun bo jak to pojdzie w swiat to wypierdola Twoje CV razem z Twoim kodem